### DISCUSSION TITLE
The issue about two action use same one SaveOperation but with validate issue, and how?

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1892] [Q&A] [answered]
Please check following code:

I have a `SaveUser` operation

```crystal
class SaveUser < User::SaveOperation
  attribute password : String
  attribute password_confirmation : String

  before_save do
    validate_required password
    validate_confirmation_of password, with: password_confirmation
  end
end

```

with two action use it.

```crystal
class User::Htmx::Editable < HtmxAction
  put "/users/:id/editable" do
    user = UserQuery.find(params.get(:id))

    SaveUser.update!(user, is_editable: !user.is_editable)

    plain_text "ok"
  end
end
```

And

```crystal
class User::Htmx::Password < HtmxAction
  put "/users/:id/password" do
    user = UserQuery.find(params.get(:id))
    SaveUser.update!(
      user,
      password: params.get(:password),
      password_confirmation: params.get(:password_confirmation)
    )

    plain_text "ok"
  end
end
```

When i first add the `editable` action, it works.

Then when i add the `password` action with validate, the editable action not work anymore, with following error:

```
{"failed_to_save":"SaveUser","validation_errors":"password required"}}
```

One of the solution is, split the `SaveUser` into two individual file,  like `SaveUserPassword` and `ToggleUserEditable`, but, my question here is, like this small save_user operation, is there another pratical way to workaround this?

Thanks.

**Comment by zw963** [comment]
I solve this issue with pass a `action` attribute from action into operation, probably not good, but it is works!

**Reply by jwoertink** [reply, marked as answer]
I think this is what I would do, actually. Alternatively, you can just use `User::SaveOperation.update!(user, is_editable: !user.is_editable)` which will circumvent any callbacks and keep the SaveUser to just updating the password
